### ISSUE TITLE
Broken org.rust.cargo.fetch.actual.stdlib.metadata since 1.52

### ISSUE THREAD

**Issue by ralfbiedert** [intellij-rust/intellij-rust#7212]
## Environment

* **IntelliJ Rust plugin version:** 0.3.146.3826-211
* **Rust toolchain version:** null
* **IDE name and version:** CLion 2021.1.1 (CL-211.7142.21)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

I updated to Rust 1.52 yesterday night. This morning when opening my project at least "dot"-autocompletion stopped working:

![image](https://user-images.githubusercontent.com/301288/117433264-9862d300-af2b-11eb-92a4-2275e07e4a0b.png)

Seemingly caused by `org.rust.cargo.fetch.actual.stdlib.metadata` producing this error:

![image](https://user-images.githubusercontent.com/301288/117433752-1de68300-af2c-11eb-89a9-8abf01bbcbdf.png)

## Steps to reproduce

- Use Rust 1.52
- Run with experimental option `org.rust.cargo.fetch.actual.stdlib.metadata` enabled

Disabling that option makes autocomplete work again.

**Comment by ortem** [comment]
@ralfbiedert Could you please check the toolchain and standard library paths in `Preferences | Languages & Frameworks | Rust` settings? Are they specified and valid?

**Comment by ortem** [comment]
@ralfbiedert Thank you, paths look fine indeed.

Could you please attach your `idea.log` using `Help | Show Log in Explorer` action? (see [this instruction](https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) for more information)

**Comment by ralfbiedert** [comment]
Ah, I can see an exception in that log:

```
java.util.concurrent.CompletionException: java.lang.IllegalStateException: `cargo metadata` reported a package which does not exist at `C:\Users\rb\AppData\Local\JetBrains\CLion2021.1\intellij-rust\stdlib\1.52.0-64bf6ced63ab650e19f4ba9f1a06c67bb067418d\vendor\rustc-std-workspace-core\Cargo.toml`
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at org.rust.cargo.project.model.impl.CargoSyncTask.run(CargoSyncTask.kt:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:998)
        ...
```

[**Full idea.log**](https://github.com/intellij-rust/intellij-rust/files/6461206/idea.log)

**Comment by ortem** [comment]
@ralfbiedert Thank you!

I've only managed to get this exception after manually removing `...\vendor\rustc-std-workspace-core` folder before `Refresh Cargo Projects`. Could you please check if this directory exists on your machine? Also, you can try to manually remove the `..\CLion2021.1\intellij-rust\stdlib` folder and invoke `Refresh Cargo Projects` again (the new `stdlib` folder will be created automatically during the refresh).

**Comment by ralfbiedert** [comment]
Ok, so right now I've been running with `...fetch.actual.stdlib.metadata` disabled. I checked the folder `C:\Users\rb\AppData\Local\JetBrains\CLion2020.1\intellij-rust`, which happens to contain only:

![image](https://user-images.githubusercontent.com/301288/117977208-dea3a200-b330-11eb-9569-77ed436102cd.png)

I've then re-enabled the option, restarted CLion, and still got the same IDE error, but no other folder inside `...\JetBrains\CLion2020.1\intellij-rust` was created. 

In other words, there is no `stdlib` folder for me to remove, despite several restarts and "Refresh Cargo Projects" and toggling that option on and off. 

Interestingly enough, with `...fetch.actual.stdlib.metadata` disabled it still says "Cargo: failed", but auto-complete works:

![image](https://user-images.githubusercontent.com/301288/117977985-a8b2ed80-b331-11eb-9250-995f3e0045d5.png)

**Comment by ralfbiedert** [comment]
Just updated to `0.4.148.3911-211`; can confirm, this seems to be fixed. Thanks!
